lib/common/xml.c

```diff
@@ -1285,6 +1285,44 @@ write_xml_stream(const xmlNode *xml, const char *filename, FILE *stream,
     return rc;
 }
 
+/*!
+ * \internal
+ * \brief Write XML to a file descriptor
+ *
+ * \param[in]  xml       XML to write
+ * \param[in]  filename  Name of file being written (for logging only)
+ * \param[in]  fd        Open file descriptor corresponding to \p filename
+ * \param[in]  compress  If \c true, compress XML before writing
+ * \param[out] nbytes    Number of bytes written (can be \c NULL)
+ *
+ * \return Standard Pacemaker return code
+ */
+int
+pcmk__xml_write_fd(const xmlNode *xml, const char *filename, int fd,
+                   bool compress, int *nbytes)
```

Comment:
I wanted to start splitting it up sometime soon after discussing along what lines to do so -- so yeah we can do that now.

Reply:
I'm undecided on whether the dump-to-string functions should go in `xml_io.c` or not... They're not doing file I/O, but neither is `string2xml`/`pcmk__xml_parse`. I lean slightly toward "no."

Reply:
It probably does make sense to include the dump and escape functions, since they are only useful for preparing to output the XML. Similarly the parse functions are only useful to prepare input for use.

Reply:
> escape functions, since they are only useful for preparing to output the XML

We also use it in `pcmk__xe_set_content()`, which is called by `pcmk_create_xml_text_node()`. So not just for preparing to output the XML.

Reply:
That makes sense